core/shared/src/main/scala/scalaz/zio/Task.scala

```diff
@@ -0,0 +1,341 @@
+   */
+  final def effect[A](effect: => A): Task[A] = ZIO.effect(effect)
+
+  /** See [[scalaz.zio.ZIO.effectAsync]]
+   */
+  final def effectAsync[A](register: (Task[A] => Unit) => Unit): Task[A] =
+    ZIO.effectAsync(register)
+
+  /** See [[scalaz.zio.ZIO.effectAsyncMaybe]]
+   */
+  final def effectAsyncMaybe[A](register: (Task[A] => Unit) => Option[IO[Throwable, A]]): Task[A] =
+    ZIO.effectAsyncMaybe(register)
+
+  /** See [[scalaz.zio.ZIO.effectAsyncM]]
+   */
+  final def effectAsyncM[A](register: (Task[A] => Unit) => UIO[_]): Task[A] =
+    ZIO.effectAsyncM(register)
+
+  /** See [[scalaz.zio.ZIO.effectAsyncInterrupt]]
+   */
+  final def effectAsyncInterrupt[A](register: (Task[A] => Unit) => Either[Canceler, Task[A]]): Task[A] =
+    ZIO.effectAsyncInterrupt(register)
+
+  /** See [[scalaz.zio.ZIO.effectTotal]]
+   */
+  final def effectTotal[A](effect: => A): UIO[A] = ZIO.effectTotal(effect)
+
+  /** See [[scalaz.zio.ZIO.effectTotalWith]]
+   */
+  final def effectTotalWith[A](effect: Platform => A): UIO[A] = ZIO.effectTotalWith(effect)
+
+  /** See [[scalaz.zio.ZIO.fail]]
+   */
+  final def fail(error: Throwable): IO[Throwable, Nothing] = ZIO.fail(error)
+
+  /** See [[scalaz.zio.ZIO.flatten]]
+   */
+  final def flatten[A](zio: Task[Task[A]]): Task[A] =
+    ZIO.flatten(zio)
+
+  /** See [[scalaz.zio.ZIO.foldLeft]]
+   */
+  final def foldLeft[S, A](in: Iterable[A])(zero: S)(f: (S, A) => Task[S]): Task[S] =
+    ZIO.foldLeft(in)(zero)(f)
+
+  /** See [[scalaz.zio.ZIO.foreach]]
+   */
+  final def foreach[A, B](in: Iterable[A])(f: A => Task[B]): Task[List[B]] =
+    ZIO.foreach(in)(f)
+
+  /** See [[scalaz.zio.ZIO.foreachPar]]
+   */
+  final def foreachPar[A, B](as: Iterable[A])(fn: A => Task[B]): Task[List[B]] =
+    ZIO.foreachPar(as)(fn)
+
+  /** See [[scalaz.zio.ZIO.foreachParN]]
+   */
+  final def foreachParN[A, B](n: Long)(as: Iterable[A])(fn: A => Task[B]): Task[List[B]] =
+    ZIO.foreachParN(n)(as)(fn)
+
+  /** See [[scalaz.zio.ZIO.foreach_]]
+   */
+  final def foreach_[A](as: Iterable[A])(f: A => Task[_]): Task[Unit] =
+    ZIO.foreach_(as)(f)
+
+  /** See [[scalaz.zio.ZIO.foreachPar_]]
+   */
+  final def foreachPar_[A, B](as: Iterable[A])(f: A => Task[_]): Task[Unit] =
+    ZIO.foreachPar_(as)(f)
+
+  /** See [[scalaz.zio.ZIO.foreachParN_]]
+   */
+  final def foreachParN_[A, B](n: Long)(as: Iterable[A])(f: A => Task[_]): Task[Unit] =
+    ZIO.foreachParN_(n)(as)(f)
+
+  /** See [[scalaz.zio.ZIO.forkAll]]
+   */
+  final def forkAll[A](as: Iterable[Task[A]]): UIO[Fiber[Throwable, List[A]]] =
+    ZIO.forkAll(as)
+
+  /** See [[scalaz.zio.ZIO.forkAll_]]
+   */
+  final def forkAll_[A](as: Iterable[Task[A]]): UIO[Unit] =
+    ZIO.forkAll_(as)
+
+  /** See [[scalaz.zio.ZIO.fromFunction]]
+   */
+  final def fromFunction[A](f: Any => A): UIO[A] =
+    ZIO.fromFunction(f)
+
+  /** See [[scalaz.zio.ZIO.fromFunctionM]]
+   */
+  final def fromFunctionM[A](f: Any => IO[Throwable, A]): Task[A] =
```

Comment:
This one only makes sense when `R` != `Any`, and it should be deleted from `Task`.